pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +395,83 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
 	_, err = m.deps.Clientset.PingcapV1alpha1().TidbClusters(tc.Namespace).Patch(tc.Name, types.MergePatchType, mergePatch)
 	return err
 }
+
+func (m *MonitorManager) smoothMigrationToStatefulSet(monitor *v1alpha1.TidbMonitor) (bool, error) {
+	// determine whether there is an old deployment
+	deploymentName := GetMonitorObjectName(monitor)
+	exist, err := m.deps.TypedControl.Exist(client.ObjectKey{
```

Comment:
Just use the deploymentLister in `m.deps` to check.

Reply:
Why it's better? This will do API call to kube-apiserver each time which is not a good practice especially in a large-scale Kubernetes cluster where kube-apiserver may already be under pressure.